src/controllers/csi/provisioner/controller.go

```diff
@@ -79,6 +82,24 @@ func (provisioner *OneAgentProvisioner) SetupWithManager(mgr ctrl.Manager) error
 func (provisioner *OneAgentProvisioner) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
 	log.Info("reconciling DynaKube", "namespace", request.Namespace, "dynakube", request.Name)
 
+	result, err := provisioner.provision(ctx, request)
+	if err != nil || !result.IsZero() {
+		return result, err
+	}
+
+	result, err = provisioner.collectGarbage(ctx, request)
+	if err != nil || !result.IsZero() {
+		return result, err
+	}
+
+	return reconcile.Result{}, nil
+}
```

Comment:
I think it would make more sense to do the 
```	if !dk.NeedsCSIDriver() {
		log.Info("CSI driver not needed")
		return reconcile.Result{RequeueAfter: longRequeueDuration}, provisioner.db.DeleteDynakube(ctx, request.Name)
	}
```
check here, and also the `dk, err := provisioner.getDynaKube(ctx, request.NamespacedName)`

and then the `provision` and `gc` doesn't need to care about the `reconcile.Request`

furthermore I would handle all possible `reconcile.Result{}` in this function, so the provisioner/gc functions would be easier to understand/test
the  `reconcile.Result{}` should be something like
- longDuration if `!dk.NeedsCSIDriver()`
- defaultDuration otherwise